### PR TITLE
Fix: ignore undefined/null references when resolving

### LINKED ISSUE #61: schema resolveFor needs to work with undefined/null references as well
Sometimes a reference is optional. In that case we need to resolve immediatly and without trying to call the item

https://github.com/Hypheme/harmonized.js/blob/master/src/Schema.js#L327
https://github.com/Hypheme/harmonized.js/blob/master/src/Schema.js#L332

### PR DESCRIPTION
**Pull request by tchock**
Closes #61

### CODE REVIEW

File: src/Schema.js [hunk 1]
```diff
@@ -326,13 +326,17 @@ class Schema {
     const unresolvedReferences = [];
     this.references.forEach((value, key) => {
       const ref = get(item, key);
+      console.log(key, ref);
```
Comment (JohannesMerz) [inline src/Schema.js]: ditch that

File: src/Schema.js [hunk 2]
```diff
@@ -326,13 +326,17 @@ class Schema {
...
+      if (ref === undefined || ref === null) {
+        return;
+      }
+
+      console.log('after this');
```
Comment (JohannesMerz) [inline src/Schema.js]: and this

File: src/Schema.js [hunk 3]
```diff
@@ -326,13 +326,17 @@ class Schema {
     const unresolvedReferences = [];
     this.references.forEach((value, key) => {
       const ref = get(item, key);
+      console.log(key, ref);
+      if (ref === undefined || ref === null) {
```
Comment (JohannesMerz) [inline src/Schema.js]: they are mobx observables - not sure if that works then but we'll see i guess

**Review by JohannesMerz — CHANGES_REQUESTED** [review]

**Review by JohannesMerz — APPROVED** [review]

File: src/Schema.js (hunk 3 above)
Comment (tchock) [reply src/Schema.js]: they will work, mobx observables return undefined or null :)

Comment (JohannesMerz) [reply src/Schema.js]: ah sweet, aproved then

**Pull request merged**